scripts-available/CDB_TableMetadata.sql

```diff
@@ -120,29 +120,20 @@ BEGIN
   END IF;
 
   -- Cleanup stale entries
-  DELETE FROM public.CDB_TableMetadata
-   WHERE NOT EXISTS (
-    SELECT oid FROM pg_class WHERE oid = tabname
-  );
-
-  WITH nv as (
-    SELECT TG_RELID as tabname, NOW() as t
-  ), all_dependents as (
-    SELECT nv.tabname, nv.t
-    FROM nv
-    UNION
-    SELECT
-      dv.dependent_oid as tabname,
-      nv.t
-    FROM nv, public.CDB_TableMetadata_DependentViews(ARRAY[nv.tabname::regclass]) dv
-  ), updated as (
-    UPDATE public.CDB_TableMetadata x SET updated_at = ad.t
-    FROM all_dependents ad WHERE x.tabname = ad.tabname
-    RETURNING x.tabname
-  )
-  INSERT INTO public.CDB_TableMetadata SELECT ad.*
-  FROM all_dependents ad LEFT JOIN updated USING(tabname)
-  WHERE updated.tabname IS NULL;
+  DELETE FROM cdb_tablemetadata m
+  USING (
+     SELECT tabname
+     FROM cdb_tablemetadata
+     WHERE NOT EXISTS (
+       SELECT oid FROM pg_class WHERE oid = tabname
+     )
+     ORDER BY tabname ASC
+     FOR UPDATE
+  ) d
+  WHERE d.tabname = m.tabname;
+
+  -- Update metadata for table
+  PERFORM public.CDB_TableMetadataTouch(TG_RELID::regclass);
```

Comment:
I see current prod entry for TG_RELID as -
IF TG_RELID = 'cartodb.CDB_TableMetadata'::regclass::oid THEN
    RETURN NULL;
See the schema name difference.. It is possible that Carto team changed that to public.. We certainly have it under schema - cartodb

Reply:
It should work ok..

Reply:
The extension used to be under the `public` schema but was later moved to `cartodb` I believe.  For legacy reasons and potentially for compatibility with deployments still under `public`, the code still references the public schema.  There is logic in the Makefile to replace references to public with cartodb.
https://github.com/bloomberg/cartodb-postgresql/blob/7db201a2a8843e90c9ab16dded276be4bbaf8b27/Makefile#L116-L123

Reply:
Yes!